packages/template-retail-react-app/app/pages/checkout/partials/shipping-multi-address.jsx

```diff
@@ -492,23 +506,11 @@ const ShippingMultiAddress = ({
                                             </Text>
 
                                             <Box w="100%" mb={6}>
-                                                <Select
-                                                    value={
-                                                        showAddAddressForm[addressKey]
-                                                            ? ADD_NEW_ADDRESS_OPTION_VALUE
-                                                            : selectedAddressId || ''
-                                                    }
-                                                    onChange={(e) => {
-                                                        const value = e.target.value
-
-                                                        if (
-                                                            value === ADD_NEW_ADDRESS_OPTION_VALUE
-                                                        ) {
-                                                            setShowAddAddressForm((prev) => ({
-                                                                ...prev,
-                                                                [addressKey]: true
-                                                            }))
-                                                        } else {
+                                                <VStack spacing={3} align="stretch">
+                                                    <Select
+                                                        value={selectedAddresses[addressKey] || ''}
+                                                        onChange={(e) => {
+                                                            const value = e.target.value
                                                             setShowAddAddressForm((prev) => ({
                                                                 ...prev,
                                                                 [addressKey]: false
```

Comment:
isn't this supposed to be an object? why are we setting a boolean here? If addressed is not selected, let's use `.filter` to remove unselected key id

Reply:
nvm, sorry I misunderstood this function